civihr_employee_portal/tdd/HelperFunctionsTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+require_once __DIR__ . '/../views/civihr_employee_portal.views.inc';
+
+class HelperFunctionsTest extends PHPUnit_Framework_TestCase
```

Comment:
Agree that it has a lot of problems. I'm not sure how to go about testing code that is an extension of a framework. I think writing Symfony bundles (which is similar to extensions) each bundle had it's own composer dependencies so it would have all the required code even if just testing outside of the framework.

Regarding the `get_where_parts` function visibility. It's common to have methods that don't make sense being public but that you still want to test. I think if we had proper classes here I would have no problem in making a utility class for dealing with queries. I'm in favor of smaller classes with very specific purpose.